### ISSUE TITLE
simple-keyvalue-client crashes non-deterministally

### ISSUE THREAD

**Issue by mitjat** [oasislabs/oasis-core#2531]
#### SUMMARY
`simple-keyvalue-client` (source: tests/clients/simple-keyvalue/src/main.rs) is a simple binary that exercises the default network fixture erected by `oasis-net-runner`. It does so by invoking a series of hard-coded RPCs on the `simple-keyvalue` runtime. It's very much like an e2e test. Sometimes (in my testing, 2 out of 4 times), it crashes.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
tests/clients/simple-keyvalue

##### OASIS NODE VERSION
```paste below
Software version: 19.0-git7c3d5f3c
Runtime protocol version: 0.10.0
Consensus protocol version: 0.22.0
Committee protocol version: 0.7.0
Tendermint core version: 0.32.8
ABCI library version: 0.16.1
```

##### OS / ENVIRONMENT
$ go version
go version go1.13.5 linux/amd64
$ rustc --version
rustc 1.39.0-nightly (521d78407 2019-08-25)

##### STEPS TO REPRODUCE
Follow steps from oasis-core README.md:
```
./go/oasis-net-runner/oasis-net-runner \
  --net.node.binary go/oasis-node/oasis-node \
  --net.runtime.binary target/debug/simple-keyvalue \
  --net.runtime.loader target/debug/oasis-core-runtime-loader \
  --net.keymanager.binary target/debug/oasis-core-keymanager-runtime

./target/debug/simple-keyvalue-client \
  --runtime-id 0000000000000000000000000000000000000000000000000000000000000000 \
  --node-address unix:/tmp/oasis-net-runner<REPLACE_THIS_WITH_YOUR_VALUE>/net-runner/network/client-0/internal.sock
```

(Note: It's not OK to have an existing instance of the network fixture already running, or a different crash will occur. Start oasis-net-runner from scratch.)

##### ACTUAL RESULTS
```paste below
[...]
Querying transaction tags (kv_op=insert)...                                                                                                                                                                
Found transaction: index=0 input=TxnCall { method: "insert", args: Map({Text("key"): Text("hello_key"), Text("value"): Text("hello_value")}) } output=Success(Null)                                        
Querying transaction tags (kv_op=insert)...                                                                                                                                                                
thread 'main' panicked at 'query transactions: CallFailed("Codec(ErrorImpl { code: Message(\"invalid type: null, expected a sequence\"), offset: 0 })")                                                    
```

When running with backtrace, line 209 from tests/clients/simple-keyvalue/src/main.rs is the last identifiable problatic line:
```
199     println!("Querying transaction tags (kv_op=insert)...");
200     let query = Query {
201         round_min: 0,
202         round_max: 3,
203         conditions: vec![QueryCondition {
204             key: b"kv_op".to_vec(),
205             values: vec![b"insert".to_vec().into()],
206         }],
207         limit: 0,
208     };
209     let txns = rt
210         .block_on(kv_client.txn_client().query_txs(query))
211         .expect("query transactions");
212     println!("Found transactions:");
```

##### EXPECTED RESULTS
Client should finish without crashing. It did, 2/4 times.

**Comment by kostko** [comment]
Actually this also seems to happen sometimes during E2E tests.

**Comment by kostko** [comment]
Should be fixed by #2542.
